### ISSUE TITLE
Indexing queue stuck on Idle

### ISSUE THREAD

**Issue by nathanielks** [algolia/algoliasearch-wordpress#495]
### What did you expect to happen?

The queue remain in the "Running" state, posts be indexed, and have the results displayed in Algolia's admin.

### What happened instead?

The queue runs for a moment, then switched back to "Idle" with a pending task in the queue.

### How can we reproduce this behavior?

Unfortunately, I'm not entirely sure. I installed this in a production-like environment (container duplicate). We're running WooCommerce and SearchWP. I've disabled all plugins save for Algolia and reverted to a stock WP theme and the issue persists.

### Technical info
* WordPress version: 4.7.0
* Algolia Search plugin version: 1.7.0

Here's the logs as well:
```
57354	operation	2017/01/23 at 06:17:02	[1] Pushed settings to index 'wp_posts_post_tmp'.
57353	operation	2017/01/23 at 06:17:02	[1] Cleared index 'wp_posts_post_tmp'.	
57352	info	2017/01/23 at 06:17:02	About to run 1 task.	
57351	info	2017/01/23 at 06:17:01	New task re_index_items was queued.
```

**Comment by nathanielks** [comment]
Hitting the "Process Queue" button doesn't do anything either.

**Comment by nathanielks** [comment]
Rather, it doesn't help. It just repeats the same behavior described above.

**Comment by nathanielks** [comment]
Excuse me, I thought PHP error logs were empty but it appears there's an issue with `is_file()`.

**Comment by nathanielks** [comment]
Here's a gist of what we have of the log, unfortunately it's not too much but it's something: 

https://gist.github.com/nathanielks/2bc6a2435d7d603756967d0bce06383c

It appears `Algolia_Posts_Index` and `Algolia_Searchable_Posts_Index` are calling the [`simple_html_dom`](https://github.com/algolia/algoliasearch-wordpress/blob/9519a55d0a41e6f14942e7afe1d758d16e059162/includes/libraries/php-dom-parser/lib/simple_html_dom.php#L1022) and it's trying to find a file with the same name as the entirety of the post's content and it doesn't like that?

**Comment by rayrutjes** [comment]
This is definitely a bug in the underlying dom lib. I opened a ticket here: https://github.com/algolia/php-dom-parser/issues/9 and will make sure to ship a fix as fast as possible.

In the meantime, you can comment some lines in that lib by following this example: https://github.com/algolia/php-dom-parser/pull/10

Let me know if that works.

**Comment by rayrutjes** [comment]
This has now been merged in develop and will be part of 1.8.0 release.
